GVFS/GVFS/CommandLine/HealthVerb.cs

```diff
@@ -0,0 +1,246 @@
+                string percent = this.FormatPercent(pair.Item2);
+                this.Output.WriteLine(" " + percent + " | " + dir);
+            }
+
+            bool healthyRepo = (enlistmentHealthData.PlaceholderPercentage + enlistmentHealthData.ModifiedPathsPercentage) < MaximumHealthyHydration;
+
+            this.Output.WriteLine("\nRepository status: " + (healthyRepo ? "Healthy" : "Unhealthy"));
+        }
+
+        /// <summary>
+        /// Takes a fractional decimal and formats it as a percent taking exactly 4 characters with no decimals
+        /// </summary>
+        /// <param name="percent">Fractional decimal to format to a percent</param>
+        /// <returns>A 4 character string formatting the percent correctly</returns>
+        private string FormatPercent(decimal percent)
+        {
+            return percent.ToString("P0").PadLeft(4);
+        }
+
+        /// <summary>
+        /// Parse a line of the git index coming from the ls-tree endpoint in the git process to get the path to that file
+        /// </summary>
+        /// <param name="line">The line from the output of the git index</param>
+        /// <returns>The path extracted from the provided line of the git index</returns>
+        private string TrimGitIndexLine(string line)
+        {
+            return line.Substring(line.IndexOf('\t') + 1);
+        }
+
+        /// <summary>
+        /// Talk to the mount process across the named pipe to get a list of the modified paths
+        /// </summary>
+        /// <param name="enlistment">The enlistment being operated on</param>
+        /// <returns>An array containing all of the modified paths in string format</returns>
+        private string[] GetModifiedPathsFromPipe(GVFSEnlistment enlistment)
+        {
+            using (NamedPipeClient pipeClient = new NamedPipeClient(enlistment.NamedPipeName))
+            {
+                string[] modifiedPathsList = { };
+
+                if (!pipeClient.Connect())
+                {
+                    this.ReportErrorAndExit("Unable to connect to GVFS.  Try running 'gvfs mount'");
+                }
+
+                try
+                {
+                    NamedPipeMessages.Message modifiedPathsMessage = new NamedPipeMessages.Message(NamedPipeMessages.ModifiedPaths.ListRequest, NamedPipeMessages.ModifiedPaths.CurrentVersion);
+                    pipeClient.SendRequest(modifiedPathsMessage);
+
+                    NamedPipeMessages.Message modifiedPathsResponse = pipeClient.ReadResponse();
+                    if (!modifiedPathsResponse.Header.Equals(NamedPipeMessages.ModifiedPaths.SuccessResult))
+                    {
+                        this.Output.WriteLine("Bad response from modified path pipe: " + modifiedPathsResponse.Header);
+                        return modifiedPathsList;
+                    }
+
+                    modifiedPathsList = modifiedPathsResponse.Body.Split(new char[] { '\0' }, StringSplitOptions.RemoveEmptyEntries);

```

Comment:
This `modifiedPathsList` may be very large. Instead of using `Split()` to construct an array, we will want to revisit this to parse the body dynamically and use `yield return substring;` (returning an `IEnumerable<string>` in the method).

We can save that change for after you have this working, tested, and merged.

Reply:
That makes sense, I'll add it to my list of todos!

Reply:
Or when we move the modified paths to the SQLite database you will not have to split and the query will give you the individual paths. :)